lib/meta_logger/slicer.ex

```diff
@@ -0,0 +1,46 @@
+defmodule MetaLogger.Slicer do
+  @moduledoc """
+  Responsible for slicing log entries according to the given the max length option.
+  """
+
+  @typedoc "Max length in bytes or `:infinity` if the entry should not be sliced."
+  @type max_entry_length :: non_neg_integer() | :infinity
+
+  @doc """
+  Returns sliced log entries sccording to the given max length option.
+
+  If the entry is smaller than given max length, or if `:infinity ` is given
+  as option, a list with one entry is returned. Otherwise a list with multiple
+  entries is returned.
+
+  ## Examples
+
+      iex> #{inspect(__MODULE__)}.slice("1234567890", 10)
+      ["1234567890"]
+
+      iex> #{inspect(__MODULE__)}.slice("1234567890", :infinity)
+      ["1234567890"]
+
+      iex> #{inspect(__MODULE__)}.slice("1234567890", 5)
+      ["12345", "67890"]
+
+  """
+  @spec slice(String.t(), max_entry_length()) :: [String.t()]
+  def slice(entry, max_entry_length)
+      when max_entry_length == :infinity
+      when byte_size(entry) <= max_entry_length,
+      do: [entry]
```

Comment:
What do you think if to make `:infinity` as a default value as well?

Reply:
It is, check the previous lines. We set the default options already